src/components/hack-details.js

```diff
@@ -1,100 +1,174 @@
-      justify="center"
-      alignItems="stretch"
-      alignContent="center"
-    >
-      <Paper className={classes.root}>
-        <Grid
-          container
-          justify="space-between"
-          alignItems="center"
-          alignContent="center"
-          spacing={1}
-        >
-          <Grid item xs={10}>
-            <Chip
-              icon={<ErrorOutlineOutlinedIcon />}
-              label="status"
-              color="secondary"
-            />
-          </Grid>
-          <Grid item xs={2} className={classes.rightFeild}>
-            <Button variant="contained" color="primary">
-              Join Hack
-            </Button>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h4">{title}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Idea:</Typography>
-          </Grid>
-          <Grid item xs={10}>
-            <Typography variant="body1">{description}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Goal:</Typography>
-          </Grid>
-          <Grid item xs={10}>
-            <Typography variant="body1">{goal}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Team:</Typography>
+  const joinHack = event => {
+    const object = {
+      hackId: props.match.params.id,
+      userId: props.userId
+    };
+    Axios.post(process.env.REACT_APP_API_URL + "joinhack", object).then(res => {
+      setTeam(res.data.team);
+    });
+  };
+
+  const joinHackButton = () => {
+    if (team !== undefined) {
+      if (team.includes(props.userId)) {
+        return (
+          <Button
+            variant="contained"
+            color="primary"
+            onClick={joinHack}
+            disabled
+          >
+            Join Hack
+          </Button>
+        );
+      } else {
+        return (
+          <Button variant="contained" color="primary" onClick={joinHack}>
+            Join Hack
+          </Button>
+        );
+      }
+    }
+    return (
+      <Button variant="contained" color="primary" onClick={joinHack}>
+        Join Hack
+      </Button>
+    );
+  };
+
+  if (title !== "") {
+    return (
+      <Grid
+        data-testid="main-container"
+        container
+        direction="column"
+        justify="center"
+        alignItems="stretch"
+        alignContent="center"
+      >
+        <Paper className={classes.root}>
+          <Grid
+            container
+            justify="space-between"
+            alignItems="center"
+            alignContent="center"
+            spacing={1}
+          >
+            <Grid item xs={9}>
+              <Chip
+                icon={<ErrorOutlineOutlinedIcon />}
+                label="status"
+                color="secondary"
+              />
+            </Grid>
+            <Grid item xs={3} className={classes.rightFeild}>
+              {joinHackButton()}
```

Comment:
Here is another function that should be turned into a component.

```
<JoinHackButton team={team} />
```

Reply:
Join button has been made into a component